nexus/analyzer/src/lib.rs

```diff
@@ -808,10 +800,11 @@ fn parse_db_options(
     Ok(config)
 }
 
-fn parse_metadata_db_info(conn_str: &str) -> anyhow::Result<Option<PostgresConfig>> {
-    if conn_str.is_empty() {
-        return Ok(None);
-    }
+fn parse_metadata_db_info(conn_str: Option<String>) -> anyhow::Result<Option<PostgresConfig>> {
```

Comment:
```suggestion
fn parse_metadata_db_info(conn_str: Option<&str>) -> anyhow::Result<Option<PostgresConfig>> {
```
Then remove `map(|s| s.to_string())` from callers